compiler/back_end/util/code_template_test.py

```diff
@@ -14,85 +14,85 @@
 
 """Tests for code_template."""
 
+import string
 import unittest
 from compiler.back_end.util import code_template
 
+def _format_template_str(template: str, **kwargs) -> str:
+  return code_template.format_template(string.Template(template), **kwargs)
 
 class FormatTest(unittest.TestCase):
   """Tests for code_template.format."""
 
   def test_no_replacement_fields(self):
-    self.assertEqual("foo", code_template.format_template("foo"))
-    self.assertEqual("{foo}", code_template.format_template("{foo}"))
-    self.assertEqual("$foo$", code_template.format_template("$foo$"))
-    self.assertEqual("$_foo$", code_template.format_template("$_foo$"))
-    self.assertEqual("$foo_$", code_template.format_template("$foo_$"))
+    self.assertEqual("foo", _format_template_str("foo"))
+    self.assertEqual("{foo}", _format_template_str("{foo}"))
```

Comment:
If the string `${foo}` is needed then I think that's

```
self.assertEqual("{foo}", _format_template_str("{foo}"))
```

Based on https://docs.python.org/3/library/string.html#template-strings . The extra test would be useful to inform readers who need that escaping.

Reply:
This maybe didn't render correctly but I think you want me to test `$${foo}`, if so added that.